syntax = "proto3";

package is;

import "is/msgs/validate.proto";
import "is/msgs/image.proto";

message SkeletonsGrouperOptions {
  string broker_uri = 1;
  string zipkin_host = 2;
  uint32 zipkin_port = 3;
  // Unique id that identifies a group of differente sources used
  // to group and localize skeletons detections
  int32 id = 11[(is.validate.rules).int32 = {gte: 0}];
  // Ids of the cameras that will be used for group skeletons detections
  map<int64, is.vision.BoundingPoly> cameras = 4;
  // Referential used for generate the output localizations
  int64 referential = 5;
  // Minimum error in pixels accepted for the proccess of groupping
  double min_error = 6;
  // Minimum score eligible for incoming joints
  double min_score = 9 [(is.validate.rules).double = {gte: 0.0, lte: 1.0}];
  // Maximum joint-joint distance acceptable. Must be in the same unit as extrinsic parameters.
  double max_distance = 10 [(is.validate.rules).double = {gt: 0.0}];
  // Period in milliseconds of the rate that detections will be groupped
  int32 period_ms = 7 [(is.validate.rules).int32 = {gte: 50, lte: 250}];
  // Corresponds the number of samples that, in case of 
  // not receive a message from an especific camera, 
  // the buffer from this camera will be released
  int32 release_samples = 8 [(is.validate.rules).int32 = {gte: 1, lte: 20}];
}